include/sys/zfs_context.h

```diff
@@ -123,6 +123,7 @@ extern "C" {
 #define	noinline	__attribute__((noinline))
 #define	likely(x)	__builtin_expect((x), 1)
 #define	unlikely(x)	__builtin_expect((x), 0)
+#define	__maybe_unused	__attribute__((unused))
```

Comment:
This should already be defined for user space in `lib/libspl/include/sys/debug.h` which is included above.  Are we somehow including a different `<sys/debug.h>` and that's why this was needed?

Reply:
Ah, no, sorry- this was preemptive because I have a local __maybe_unused defined in system headers that I was pretty sure was polluting somehow, and ENOCRITICALTHINKING :-) I'll drop it here soon.